SharpGenTools.Sdk/Extensibility/DefaultExtensionAssemblyLoader.Core.cs

```diff
@@ -0,0 +1,45 @@
+﻿﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#if NETCOREAPP
+
+ using System.Diagnostics;
+ using System.Reflection;
+ using System.Runtime.Loader;
+
+ namespace SharpGenTools.Sdk.Extensibility
+{
+    internal class DefaultExtensionAssemblyLoader : ExtensibilityAssemblyLoader
+    {
+        private AssemblyLoadContext _loadContext;
+
+        protected override Assembly LoadFromPathImpl(string fullPath)
+        {
+            //.NET Native doesn't support AssemblyLoadContext.GetLoadContext. 
+            // Initializing the _loadContext in the .ctor would cause
+            // .NET Native builds to fail because the .ctor is called. 
+            // However, LoadFromPathImpl is never called in .NET Native, so 
+            // we do a lazy initialization here to make .NET Native builds happy.
```

Comment:
If this file isn't lifted directly from Roslyn, we can simplify some of it since we don't need to handle .NET Native scenarios.

Reply:
I'll see what I can do. This file originated from Roslyn, but in general files in `Extensibility` don't have to follow Roslyn upstream exactly, so that simplifications can be made.